### ISSUE TITLE
Add JSON schema for variant files

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/rattler-build#295]
Similar to #281 

It would be nice to have an up-to-date JSON schema for the variant files.

**Comment by wolfv** [comment]
Agreed, that would be nice. I assume you know about our JSON schemas in https://github.com/prefix-dev/recipe-format ? They are generated with pydantic. I think it would be quite straightforward to make one for the variant config files.
